contracts/DataAvailability.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {AccessControlDefaultAdminRules} from '@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol';
+
+contract DataAvailability is AccessControlDefaultAdminRules {
+    /**
+     * @dev DataAvailability stores information about the objective function value for users
+     * across timestamps.
+     * 
+     * The contract creator and any permitted uploaders may upload user objective function data
+     * at a given timestamp. The objective function data that are uploaded represent the delta
+     * between the objective function value at the current timestamp and the value at the
+     * previous timestamp that data was uploaded. The data is not actually stored in the contract,
+     * but events are emitted for each upload, which can be used to reconstruct the full data history.
+     *
+     * Instead of storing all data in the contract, a rolling hash of the data is calculated and stored
+     * for each timestamp. When verifying the data, the submitted proof will contain a hash of the data
+     * computed in a trusted zkVM; if the hash contained in the proof matches the hash stored in the contract,
+     * the data is considered valid.
```

Comment:
What would happen if the hash stored in the contract and the hash contained in the proof don't match and the data is now considered invalid?